### PR TITLE
Improve CLI output

### LINKED ISSUE #384: [linter] Pretty print checks
Currently, the checks are written as one word without any way to differentiate between words. 

I'd consider this a blocking easy for the first official release.

### PR DESCRIPTION
**Pull request by ithinkihaveacat**
Output changes from e.g.

ampimgamppixelpreferred (PASS)

to

1x1 images are specified by <amp-pixel>? (PASS)

Fixes #384 

<details>
Before:

```text
$ amplint https://amp.dev/
ampimgamppixelpreferred (PASS)

ampimgheightwidthisok (WARN)

  [https://blog.amp.dev/wp-content/uploads/2019/06/cf_hero.png]: actual ratio [1999/1140 = 1.75] does not match specified [16/9 = 1.77]

ampimgheightwidthisok (WARN)

  [https://blog.amp.dev/wp-content/uploads/2018/10/img_20180926_163001-01.jpeg]: actual ratio [3680/2314 = 1.59] does not match specified [16/9 = 1.77]

ampvideoissmall (PASS)

ampvideoisspecifiedbyattribute (PASS)

endpointsareaccessiblefromcache (PASS)

endpointsareaccessiblefromorigin (PASS)

linkrelcanonicalisok (PASS)

metacharsetisfirst (FAIL)

  <meta charset> not the first <meta> tag

runtimeispreloaded (PASS)
```

After:

```text
$ amplint https://amp.dev/
1x1 images are specified by <amp-pixel>? (PASS)

All <amp-img> have reasonable width and height? (WARN)

  [https://blog.amp.dev/wp-content/uploads/2019/06/cf_hero.png]: actual ratio [1999/1140 = 1.75] does not match specified [16/9 = 1.77]

All <amp-img> have reasonable width and height? (WARN)

  [https://blog.amp.dev/wp-content/uploads/2018/10/img_20180926_163001-01.jpeg]: actual ratio [3680/2314 = 1.59] does not match specified [16/9 = 1.77]

All videos are under 4MB? (PASS)

<amp-video><source/></amp-video> syntax is used for video? (PASS)

Endpoints are accessible from cache? (PASS)

Endpoints are accessible from origin? (PASS)

Story is self-canonical? (PASS)

Meta charset is first? (FAIL)

  <meta charset> not the first <meta> tag

Runtime is preloaded? (PASS)
```
</details>

### CODE REVIEW

File: packages/linter/src/rule.ts [hunk 1]
```diff
@@ -8,39 +8,52 @@ export interface RuleConstructor {
...
+      title:
+        this.constructor.name.replace(
+          /([a-z])([A-Z])/g,
+          (_, c1, c2) => `${c1} ${c2.toLowerCase()}`
+        ) + "?",
+      info: ""
     };
   }
+  protected async pass(s?: string) {
+    return Object.assign(
```
Comment (sebastianbenz) [inline packages/linter/src/rule.ts]: suggestion, extract 

```
return Object.assign(
       status: Status.FAIL,		       {
       message: s,		         status: Status.FAIL,
       url: Rule.href,		         message: s
       description: Rule.info		       },
     };		       this.meta()
     );
```

into a helper method

File: packages/linter/src/rule.ts [hunk 2]
```diff
@@ -8,39 +8,52 @@ export interface RuleConstructor {
...
+  }
   protected async fail(s: string) {
-    return {
-      status: Status.FAIL,
-      message: s,
-      url: Rule.href,
-      description: Rule.info
-    };
+    return Object.assign(
+      {
```
Comment (sebastianbenz) [inline packages/linter/src/rule.ts]: easier:

```
return {
  status: Status.FAIL,
  message: s
  ...this.meta(),
},
```

Comment (sebastianbenz) [reply packages/linter/src/rule.ts]: If that works in TS

File: packages/linter/src/rule.ts [hunk 3]
```diff
@@ -8,39 +8,52 @@ export interface RuleConstructor {
...
+      url: "",
+      title:
+        this.constructor.name.replace(
+          /([a-z])([A-Z])/g,
+          (_, c1, c2) => `${c1} ${c2.toLowerCase()}`
+        ) + "?",
+      info: ""
     };
   }
+  protected async pass(s?: string) {
```
Comment (sebastianbenz) [inline packages/linter/src/rule.ts]: why are these methods `async`?

**Comment by ithinkihaveacat** [comment]
Pushed to 1a125d2 to address #408.

**Comment by sebastianbenz** [comment]
One other thing. I'd rephrase the results to facts instead of questions: 

`All videos are under 4MB?` -> `Videos are under 4MB`

File: packages/linter/src/rule.ts (hunk 3 above)
Comment (ithinkihaveacat) [reply packages/linter/src/rule.ts]: It's because `run()` returns a promise--the test could be async, and often is. However, if it's not, and `pass()` was async, then you'd need to return `Promise.resolve(this.pass())` to satisfy the signature.

It does feel slightly weird though … I'm not 100% happy with the design of the linter.

**Comment by ithinkihaveacat** [comment]
Re prettifying output, how about color?

Screenshot 2019-07-22 at 22 58 28

(Could be better, I'm no expert either!)

File: packages/linter/src/rule.ts (hunk 2 above)
Comment (ithinkihaveacat) [reply packages/linter/src/rule.ts]: It does! Changed.

**Comment by ithinkihaveacat** [comment]
Thanks for your comments! Please take another look.

File: packages/linter/package.json [hunk 4]
```diff
@@ -9,6 +9,7 @@
     "@ampproject/toolbox-cache-list": "^1.0.0-beta.5",
     "@ampproject/toolbox-cache-url": "^1.0.0-beta.5",
     "amphtml-validator": "1.0.23",
+    "chalk": "^2.4.2",
```
Comment (sebastianbenz) [inline packages/linter/package.json]: pin version

File: packages/linter/src/rules/IsValid.ts [hunk 5]
```diff
@@ -9,4 +9,11 @@ export class IsValid extends Rule {
       ? this.pass()
       : this.fail(JSON.stringify(res.errors));
   }
+  meta() {
+    return {
+      url: "https://validator.ampproject.org",
```
Comment (sebastianbenz) [inline packages/linter/src/rules/IsValid.ts]: s/ampproject.org/amp.dev

**Comment by ithinkihaveacat** [comment]
That was fast! Changes made.
